lib/render/mayaToHydra/renderOverride.cpp

```diff
@@ -0,0 +1,842 @@
+        _UpdateRenderGlobals();
+    }
+}
+
+void MtohRenderOverride::SelectionChanged() { _selectionChanged = true; }
+
+void MtohRenderOverride::_SelectionChanged() {
+    if (!_selectionChanged) { return; }
+    _selectionChanged = false;
+    MSelectionList sel;
+    if (!TF_VERIFY(MGlobal::getActiveSelectionList(sel))) { return; }
+    SdfPathVector selectedPaths;
+    auto selection = boost::make_shared<HdSelection>();
+
+#if WANT_UFE_BUILD
+    const UFE_NS::GlobalSelection::Ptr& ufeSelection =
+        UFE_NS::GlobalSelection::get();
+#endif // WANT_UFE_BUILD
+
+    for (auto& it : _delegates) {
+#if WANT_UFE_BUILD
+        if (it->SupportsUfeSelection()) {
+            if (ufeSelection) {
+                it->PopulateSelectedPaths(
+                    *ufeSelection, selectedPaths, selection);
+            }
+            // skip non-ufe PopulateSelectedPaths call
+            continue;
+        }
+#endif // WANT_UFE_BUILD
+        it->PopulateSelectedPaths(sel, selectedPaths, selection);
+    }
+    _selectionCollection.SetRootPaths(selectedPaths);
+    _selectionTracker->SetSelection(HdSelectionSharedPtr(selection));
+    TF_DEBUG(HDMAYA_RENDEROVERRIDE_SELECTION)
+        .Msg(
+            "MtohRenderOverride::_SelectionChanged - num selected: %lu\n",
+            selectedPaths.size());
+}
+
+MHWRender::DrawAPI MtohRenderOverride::supportedDrawAPIs() const {
+    return MHWRender::kOpenGLCoreProfile | MHWRender::kOpenGL;
+}
+
+MStatus MtohRenderOverride::setup(const MString& destination) {
+    MStatus status;
+
+    auto panelNameAndCallbacks = _FindPanelCallbacks(destination);
+    if (panelNameAndCallbacks == _renderPanelCallbacks.end()) {
+        // Install the panel callbacks
+        MCallbackIdArray newCallbacks;
+
+        auto id = MUiMessage::add3dViewDestroyMsgCallback(
+            destination, _PanelDeletedCallback, this, &status);
+        if (status) { newCallbacks.append(id); }
+
+        id = MUiMessage::add3dViewRendererChangedCallback(
+            destination, _RendererChangedCallback, this, &status);
+        if (status) { newCallbacks.append(id); }
+
+        id = MUiMessage::add3dViewRenderOverrideChangedCallback(
+            destination, _RenderOverrideChangedCallback, this, &status);
+        if (status) { newCallbacks.append(id); }
+
+        _renderPanelCallbacks.emplace_back(destination, newCallbacks);
+    }
+
+    auto* renderer = MHWRender::MRenderer::theRenderer();
+    if (renderer == nullptr) { return MStatus::kFailure; }
+
+    if (_operations.empty()) {
+        _operations.push_back(new HdMayaSceneRender(
+            "HydraRenderOverride_Scene",
+            !_isUsingHdSt || _globals.selectionOverlay == MtohTokens->UseVp2));
+        _operations.push_back(
+            new HdMayaRender("HydraRenderOverride_Hydra", this));
+        _operations.push_back(
+            new HdMayaManipulatorRender("HydraRenderOverride_Manipulator"));
+        _operations.push_back(new MHWRender::MHUDRender());
```

Comment:
I would appreciate some doc around this section, to know easily which render op is for what. Thanks.

Reply:
Let me know if this is what you were looking for:

https://github.com/Autodesk/maya-usd/pull/231/commits/8d3612d313e34155109210727a0f4795924c10f3

Reply:
Yes. Thanks.